CMR/python/cmr/auth/token.py

```diff
@@ -0,0 +1,320 @@
+"""
+NASA EO-Metadata-Tools Python interface for the Common Metadata Repository (CMR)
+
+    https://github.com/nasa/Common-Metadata-Repository/
+
+Copyright (c) 2020 United States Government as represented by the Administrator
+of the National Aeronautics and Space Administration. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+"""
+# Library for generating EDL tokens from CMR
+# date: 2020-10-26
+# since: 0.0
+
+#import base64
+import json
+import os
+import re
+import socket
+import subprocess
+import time
+import urllib.parse
+import urllib.request
+
+import cmr.util.common as c
+
+# ##############################################################################
+# local utilities
+
+def _get_local_ip():
+    """
+    Note, this function may not always work for all users on all operating
+    systems.
+    Returns:
+        Computer's IP address, or the public IP address, or 127.0.0.1
+    """
+    try:
+        hostname = socket.gethostname()
+        ip_address = socket.gethostbyname(hostname)
+        socket.close()
+    except:
```

Comment:
Should catch a socket exception explicitly `except socket.error as msg:`

Reply:
I think so, but the API did not say it was throwing it. I'll put that in and see how it does. pylint would like me to do that.